internal/provider/resource_group.go

```diff
@@ -0,0 +1,294 @@
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	tharsis "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg"
+	ttypes "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg/types"
+)
+
+// GroupModel is the model for a group.
+type GroupModel struct {
+	ID          types.String `tfsdk:"id"`
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+	ParentPath  types.String `tfsdk:"parent_path"`
+	FullPath    types.String `tfsdk:"full_path"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+}
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = &groupResource{}
+	_ resource.ResourceWithConfigure   = &groupResource{}
+	_ resource.ResourceWithImportState = &groupResource{}
+)
+
+// NewGroupResource is a helper function to simplify the provider implementation.
+func NewGroupResource() resource.Resource {
+	return &groupResource{}
+}
+
+type groupResource struct {
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *groupResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_group"
+}
+
+// The diagnostics return value is required by the interface even though this function returns only nil.
+func (t *groupResource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	description := "Defines and manages a group."
+
+	return tfsdk.Schema{
+		Version: 1,
+
+		MarkdownDescription: description,
+		Description:         description,
+
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:                types.StringType,
+				MarkdownDescription: "String identifier of the group.",
+				Description:         "String identifier of the group.",
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+			},
+			"name": {
+				Type:                types.StringType,
+				MarkdownDescription: "The name of the group.",
+				Description:         "The name of the group.",
+				Required:            true,
+			},
+			"description": {
+				Type:                types.StringType,
+				MarkdownDescription: "A description of the group.",
+				Description:         "A description of the group.",
+				Optional:            true,
+			},
+			"parent_path": {
+				Type:                types.StringType,
+				MarkdownDescription: "Full path of the parent namespace.",
+				Description:         "Full path of the parent namespace.",
+				Required:            true,
+			},
+			"full_path": {
+				Type:                types.StringType,
+				MarkdownDescription: "The path of the parent namespace plus the name of the group.",
+				Description:         "The path of the parent namespace plus the name of the group.",
+				Computed:            true,
```

Comment:
Any values that are computed and don't change over time can use `UseStateForUnknown`